File: sem4/aids/lista2/sdkjnnf/even.c
#include<stdbool.h>
#include<stdio.h>


bool isEven(int num){
    if(num == 0) return true;
    if(num == 1) return false;
    return isEven(num - 2);
}

int main(){
    printf("%d", isEven(100000));
}

File: sem4/aids/lista2/sdkjnnf/even.go
package main

import (
	"fmt"
	"time"
)

func isEven(num int) bool {
	if num == 1 {
		return false
	} else if num == 0 {
		return true
	} else {
		return isEven(num - 2)
	}
}

func isEven2(num int) bool {
	return num%2 == 0
}

func main() {
	start := time.Now()
	fmt.Println(isEven(1000000))
	elapsed := time.Since(start)
	fmt.Println(elapsed)

	start = time.Now()
	fmt.Println(isEven2(1000000))
	elapsed = time.Since(start)
	fmt.Println(elapsed)
}
